### PR TITLE
Minor: clean up aggregates.slt tests

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Related to #4495 

# Rationale for this change

There are some out of date errors I noticed while working on https://github.com/apache/arrow-datafusion/pull/5574

# What changes are included in this PR?

uncomment some tests in aggregates.slt and remove TODOs that have been done

# Are these changes tested?
Yes they are tests

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 1]
```diff
@@ -1058,42 +1058,15 @@ NULL 4 29 1.260869565217 123 -117 23
...
+# TODO this querys output is non determinisitic (the order of the elements
+# differs run to run
+#
 # csv_query_array_agg_distinct
 # query T
 # SELECT array_agg(distinct c2) FROM aggregate_test_100
 # ----
 # [4, 2, 3, 5, 1]
 
-# TODO: aggregate_timestamps_sum
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1067]: These tests are added in https://github.com/apache/arrow-datafusion/pull/5574

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 2]
```diff
@@ -1058,73 +1058,44 @@ NULL 4 29 1.260869565217 123 -117 23
...
 # aggregate_decimal_max
-query R
-select max(c1) from d_table
+query RT
+select max(c1), arrow_typeof(max(c1)) from d_table
 ----
-110.009
+110.009 Decimal128(10, 3)
 
-# FIX: doesn't check datatype
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1115]: Added type check per comment

File: datafusion/core/tests/sqllogictests/test_files/aggregate.slt [hunk 3]
```diff
@@ -1058,73 +1058,44 @@ NULL 4 29 1.260869565217 123 -117 23
...
 query TT
 select min(t), max(t) from  (select '00:00:00' as t union select '00:00:01' union select '00:00:02')
 ----
 00:00:00 00:00:02
 
 # aggregate_decimal_min
-query R
-select min(c1) from d_table
+query RT
+select min(c1), arrow_typeof(min(c1)) from d_table
```
Comment (jackwener) [inline datafusion/core/tests/sqllogictests/test_files/aggregate.slt:1078]: 👍

**Review by jackwener — APPROVED** [review]

**Comment by jackwener** [comment]
I reviewed this PR carefully.
Because this PR isn't complex and this PR may block other PR about agg.slt(or may cause conflict), so I quickly merged it.

Thanks @alamb

**Pull request merged**